exams/pearson/audit.py

```diff
@@ -0,0 +1,272 @@
+        return re.sub(r'\s+', '', settings.EXAMS_AUDIT_ENCRYPTION_FINGERPRINT)
+
+    @property
+    def public_key(self):
+        """The public GPG key"""
+        return base64.b64decode(settings.EXAMS_AUDIT_ENCRYPTION_PUBLIC_KEY).decode('utf-8')
+
+    def get_fingerprints(self):
+        """
+        Get the fingerprints of keys currently loaded on the system
+
+        Returns:
+            list:
+                list of keys
+        """
+        return self.gpg.list_keys().fingerprints
+
+    def configure(self):
+        """
+        Performs a one-time configuration of GPG keys
+
+        Raises:
+            ImproperlyConfigured:
+                settings are missing or invalid
+
+        Returns:
+            bool:
+                True if configuration successful
+        """
+        if self.configured:
+            return
+
+        # if any of the settings are not set, we consider this feature not enabled
+        if not all(getattr(settings, key) for key in REQUIRED_SETTINGS):
+            raise ImproperlyConfigured(
+                "One of EXAMS_AUDIT_ENCRYPTION_PUBLIC_KEY or "
+                "EXAMS_AUDIT_ENCRYPTION_FINGERPRINT is required but not set"
+            )
+
+        if self.fingerprint in self.get_fingerprints():
+            self.configured = True
+            return
+
+        print(self.public_key)
+        result = self.gpg.import_keys(self.public_key)
+
+        if result.counts['not_imported'] != 0:
+            raise ImproperlyConfigured('Error importing GPG keys for exam auditing: {}'.format(result.summary()))
+
+        if self.fingerprint not in result.fingerprints:
+            raise ImproperlyConfigured(
+                "Exam auditing GPG key successfully imported but {actual} "
+                "does not contain fingerprint: {expected}".format(
+                    actual=result.fingerprints,
+                    expected=self.fingerprint,
+                )
+            )
+
+        self.configured = True
+
+    def upload_encrypted_file(self, filename, file_type):
+        """
+        Encrypt the file locally and upload it
+
+        Args:
+            filename:
+                absolute path to the local file
+            file_type (str):
+                the type of file, either RESPONSE or REQUEST
+
+        Returns:
+            str:
+                absolute path to the local encrypted file
+        """
+        encrypted_filename = '{}.gpg'.format(filename)
+
+        log.debug('Encrypting file %s to %s', filename, encrypted_filename)
+
+        try:
+            with open(filename) as source_file:
+                self.gpg.encrypt(
+                    source_file,
+                    self.fingerprint,
+                    armor=False,
+                    symmetric=False,
+                    output=encrypted_filename
+                )
+
+            return self.store.upload(encrypted_filename, file_type)
+        finally:
+            self._cleanup(encrypted_filename)
```

Comment:
I don't know if it's a recommended pattern per se, but it's mentioned in the standard docs: https://docs.python.org/3/reference/compound_stmts.html#the-try-statement
 
> When a return, break or continue statement is executed in the try suite of a try...finally statement, the finally clause is also executed ‘on the way out.’ A continue statement is illegal in the finally clause. (The reason is a problem with the current implementation — this restriction may be lifted in the future).